src/server/app/api/controllers/eventController.js

```diff
@@ -3,14 +3,19 @@ import Boom from 'boom';
 import Bounce from 'bounce';
 import { logErr } from '../../shared/utils';
 import eventStore from '../stores/eventStore';
+import userStore from '../stores/userStore';
 
 const ObjectId = mongoose.Types.ObjectId;
 
 const eventController = {
   async createEvent(req, h) {
-    let event;
+    let data;
     try {
-      event = await eventStore.createEvent(req.payload);
+      data = req.payload;
+      // Fetch user id from username and add to created.by.user
+      const user = await userStore.getUserByPhoneOrEmail(data.user.username);
```

Comment:
Is there anyway we could already have the ID of the user in the event data so that we don't have to retrieve it here? Or is that too difficult to pull off?

Reply:
We originally did have the user ID stored on the client side so we wouldn't have to do this, but I changed it so that we wouldn't have to store on the client side more information than we need. Also considered the extremely rare possibility that the user ID could be reassigned to another user, even if the user email or phone number got removed.

Reply:
> We originally did have the user ID stored on the client side so we wouldn't have to do this, but I changed it so that we wouldn't have to store on the client side more information than we need.

If we want to store as little as possible, I'd rather store user ID than username. Username is actually somewhat identifiable, and could be used in an attempt to login as another user (guess an email or something). ID is totally anonymous.

> Also considered the extremely rare possibility that the user ID could be reassigned to another user

Maybe MongoDB works differently than I'm used to, but I have always considered it a base level assumption that you never question that DB ID's always refer to the same user. It's hard to imagine it making sense for us to make decisions based on that.